src/Core/Lifecycle.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * @param {Element} element
+ * @param {Object?} state
+ */
+function lifecycleReturn (element, state) {
+	switch (typeof state) {
+		case 'object':
+			if (!state)
```

Comment:
seems like a confusing way to check `state === null`

Reply:
We want to let only `objects` and `functions` to get passed to `setState` handler but not null. How would you right this, i have a bit of bias towards switch blocks.